### PR TITLE
 Probe ingresses in kubernikus-system over region_probed_from

### PR DESCRIPTION
**Pull request by vonneudeck**
Solves Issue #213 

We probe all ingresses in kubernikus-system every five seconds with the
blackbox-exporter/prober from all the regions mentioned in the list
`region_probed_from`.
If the region the ingress is running in is part of the list it will
be treated specially and probe from inside the cluster using the
internal service instead of the ingress.

This was done to not be flooded in redundant probes while stile having
a probe from outside of the region whose number can be raised easily.
The rationale behind the handpicking of the regions listed in `region_probed_from`
is to be able to define a region that is nearby to probe from and to be flexible to add more.

This can also be used to probe other ingresses by adding the annotation `prometheus.io/probe: "true"`

### CODE REVIEW

File: charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl [hunk 1]
```diff
@@ -277,7 +279,11 @@ scrape_configs:
     replacement: ${1}://${2}${3}
     target_label: __param_target
   - target_label: __address__
+  {{- if eq . $.Values.global.region }}
     replacement: blackbox-exporter.kubernikus-system.svc:9115
+  {{- else }}
+    replacement: prober.{{ . }}.cloud.sap
```
Comment (auhlig) [inline charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: Do we want to use the prober from our bm clusters or do we want something like `prober.kubernikus.<region>.cloud.sap`?

Comment (vonneudeck) [reply charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: I am in favour of having as few blackbox-exporters as possible. It will get messy early enough and an overview of the infrastructure is already a bit daunting.
Right now, I would stick with the ones on the BM, especially as kubernikus is not rolled out everywhere yet.
If we want to move load away from the BM in the future, I would not change the URI in single configurations but rather let the DNS-entry for `prober.$region.cloud.sap` point somewhere else.

**Comment by auhlig** [comment]
The ingress spec defines [2 rules](https://github.com/sapcc/kubernikus/blob/master/charts/kube-master/templates/ingress.yaml#L14-L29), one for the api server and one for the wormhole. We would probe both using the above configuration. Could this be an alternative solution to #114?

File: charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl (hunk 1 above)
Comment (auhlig) [reply charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: Then it could be less daunting if we had the same setup in every cluster.

Comment (BugRoger) [reply charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: I would like to propose to explicitly define a (list of) proper-urls in the values of the chart. That allows us to decouple this configuration from the SAP specific setup. Additionally, we don't need to generate the proper URL here. It also opens up possibilities for completely different setups, like having SaaS hosted probers or installations in a different cloud altogether:

```
blackbox-exporters:
  - http://pingdom.com/blackbox-exporter
  - https://prober.eu-de-1.cloud.sap
  - https://proper.na-us-1.cloud.sap
  - http://blackbox-exporter.kubernikus-system.svc:9115
```

One downside of that approach is that it also requires a loop (or it will hurt with dry-coding OCD). Another idea, would be to just have:

```
blackbox-exporters:
  primary: http://pingdom.com/blackbox-exporter
  secondary: https://prober.eu-de-1.cloud.sap
```

That would allow to simplify the config and wouldn't require a golang template loop at all. With the helm double evaluation that comes with it. Extra points because it will be easy to stick it into the `extra_scrape_config` and move the whole configuration out of the Prometheus chart. That gets us one step closer to being able to use the `stable/prometheus` chart.

File: charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl [hunk 2]
```diff
@@ -288,6 +294,9 @@ scrape_configs:
     target_label: kubernetes_name
   - source_labels: [__meta_kubernetes_ingress_path]
     target_label: path
+  - target_label: region_probed_from
+    replacement: {{ . }}
+{{- end}}
```
Comment (BugRoger) [inline charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: Isn't that already clear from the proper URL? It's also another drop of special SAP sauce...

File: charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl [hunk 3]
```diff
@@ -251,7 +251,9 @@ scrape_configs:
       target_label: __metrics_path__
       replacement: /api/v1/nodes/${1}:4194/proxy/metrics
 
-- job_name: 'blackbox-ingress'
+{{- range .Values.prometheus.region_probed_from }}
```
Comment (BugRoger) [inline charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: My personal preference is to stick this config into the `extra_scrap_config` of the values. We'd have to dumb it down a bit though. Maybe even into a single global config for all installations. Like always probe from `proper.eu-nl-1.cloud.sap` and `proper.na-us-1.cloud.sap` and that's it.

File: charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl (hunk 1 above)
Comment (BugRoger) [reply charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: I'd even be happy with a single configurable prober...

File: charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl (hunk 2 above)
Comment (vonneudeck) [reply charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl]: Good point. By default the prober URI is dropped, but I will use a more generic no-SAP way to keep the information.

**Comment by vonneudeck** [comment]
I changed everything to be pulled in from`extra_scrape_config`.
